### PR TITLE
Add RabbitMQ container module

### PR DESCRIPTION
**Pull request by zteln**
Adds support for a RabbitMQ container.

### CODE REVIEW

File: lib/container/rabbitmq_container.ex [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  @default_tag "3-alpine"
+  @default_image_with_tag "#{@default_image}:#{@default_tag}"
+  @default_port 5672
+  @default_username "guest"
+  @default_password "guest"
+  @default_virtual_host "/"
+  @default_command [
+    "sh",
+    "-c",
+    "chmod 400 /var/lib/rabbitmq/.erlang.cookie; rabbitmq-server"
```
Comment (jarlah) [inline lib/container/rabbitmq_container.ex:20]: hmm this erlang cookie. Whats this?

File: lib/container/rabbitmq_container.ex [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    [
+      host: Testcontainers.get_host(),
+      port: port(container),
+      username: container.environment[:RABBITMQ_DEFAULT_USER],
+      password: container.environment[:RABBITMQ_DEFAULT_PASS],
+      virtual_host: container.environment[:RABBITMQ_DEFAULT_VHOST]
+    ]
+  end
+
+  defp virtual_host_segment(container) do
```
Comment (jarlah) [inline lib/container/rabbitmq_container.ex:204]: could probably need some documentation about what it is.

Comment (jarlah) [reply lib/container/rabbitmq_container.ex:204]: i mean i can see where its used but thats it

**Review by jarlah — COMMENTED** [review]
It looks ok. It has tests, actual usage with asserts afaik.Can you look at the comments and i can approve after whatever new commits or explanations you come with

File: lib/container/rabbitmq_container.ex (hunk 1 above)
Comment (zteln) [reply lib/container/rabbitmq_container.ex:23]: This is included in order for allowing access to CLI tools to communicate with the RabbitMQ node, which is used in the command wait strategy, with `rabbitmq-diagnostics check_running`.  I have updated the module docs  to include this. (Relevant docs: https://www.rabbitmq.com/docs/cli#erlang-cookie)

File: lib/container/rabbitmq_container.ex (hunk 2 above)
Comment (zteln) [reply lib/container/rabbitmq_container.ex:214]: I have provided docs detailing what this function is used for and how it is used in the AMQP URI scheme. (Relevant docs: https://www.rabbitmq.com/docs/uri-spec)

**Comment by zteln** [comment]
I added a commit that allows overriding of the listening port on the RabbitMQ node in the Docker container to the port given in `RabbitMQContainer.with_port/2` function.

**Comment by zteln** [comment]
@jarlah the tests seems to have failed for other reasons, could they be triggered again?

**Comment by jarlah** [comment]
im waiting for the emqx container to get ready .. will hold release before thats finished. For that reason I will let this issue lie here for a tiny bit longer before merging. But looks ok, good work 🚀 Thanks for the contribution!

**Pull request merged**